### PR TITLE
Document has checks

### PR DESCRIPTION
**Pull request by KnorpelSenf**
Documents [grammyjs/grammY#218](https://github.com/grammyjs/grammY/pull/218)

### CODE REVIEW

File: site/docs/guide/context.md [hunk 1]
```diff
@@ -83,6 +83,34 @@ bot.on("edited_message", (ctx) => {
 
 Hence, if you want to, you can forget about `ctx.message` and `ctx.channelPost` and `ctx.editedMessage` and so on and so forth, and just always use `ctx.msg` instead.
 
+## Probing via Has-checks
```
Comment (roj1512) [inline site/docs/guide/context.md:86]: I don't think we usually use a hyphen to connect nouns.

Comment (KnorpelSenf) [reply site/docs/guide/context.md:86]: Has isn't usually a noun.

Comment (roj1512) [reply site/docs/guide/context.md:86]: "Has" itself isn't a noun, but it is a noun in that context.

Comment (roj1512) [reply site/docs/guide/context.md:86]: I prefer Has Checks. The article is neither Russian to say Has-Checks, nor it is Dutch or German to say Haschecks.

Comment (KnorpelSenf) [reply site/docs/guide/context.md:86]: It is comparable to words like one-time or half-baked

Comment (KnorpelSenf) [reply site/docs/guide/context.md:86]: I'd be happy to have a different and better name if you can come up with one hehe

Comment (roj1512) [reply site/docs/guide/context.md:86]: One-time and half-baked are adjectives, and it is common to use a hyphen to form adjectives.

Comment (KnorpelSenf) [reply site/docs/guide/context.md:86]: Alright, I'll change it

File: site/docs/guide/context.md [hunk 2]
```diff
@@ -83,6 +83,34 @@ bot.on("edited_message", (ctx) => {
...
+
+::: tip Know When to Use Has Checks
+
+This is the exact same logic that is used by `bot.command("start")`.
+Note that you should usually use [filter queries](./filter-queries.md) and similar methods.
+Using Has Checks works best inside the [conversations plugin](../plugins/conversations.md).
+
+:::
+
+The Has Checks correctly narrow down the context type.
```
Comment (roj1512) [inline site/docs/guide/context.md:100]:
```suggestion
Has Checks correctly narrow down the context type.
```

Comment (roj1512) [inline site/docs/guide/context.md:100]:
```suggestion
The has checks correctly narrow down the context type.
```

File: site/docs/guide/context.md [hunk 3]
```diff
@@ -83,6 +83,34 @@ bot.on("edited_message", (ctx) => {
...
+This means that checking if a context has callback query data will tell TypeScript that the context has the field `ctx.callbackQuery.data` present.
+
+```ts
+if (ctx.hasCallbackQuery(/query-data-\d+/)) {
+  // `ctx.callbackQuery.data` is known to be present here
+  const data: string = ctx.callbackQuery.data;
+}
+```
+
+The same applies to all other Has Checks.
```
Comment (roj1512) [inline site/docs/guide/context.md:110]:
```suggestion
The same applies to all other has checks.
```

File: site/docs/guide/context.md [hunk 4]
```diff
@@ -83,6 +83,34 @@ bot.on("edited_message", (ctx) => {
...
+Check out the [API reference of the context object](https://doc.deno.land/https://deno.land/x/grammy/mod.ts/~/Context#has) to see a list of all Has Checks.
```
Comment (roj1512) [inline site/docs/guide/context.md:111]:
```suggestion
Check out the [API reference of the context object](https://doc.deno.land/https://deno.land/x/grammy/mod.ts/~/Context#has) to see a list of all has checks.
```

File: site/docs/guide/context.md [hunk 5]
```diff
@@ -83,6 +83,34 @@ bot.on("edited_message", (ctx) => {
...
+
+The context object has a few methods that allow you to probe the contained data for certain things.
+For example, you can call `ctx.hasCommand("start")` to see if the context object contains a `/start` command.
+This is why the methods are collectively named _Has Checks_.
+
+::: tip Know When to Use Has Checks
+
+This is the exact same logic that is used by `bot.command("start")`.
+Note that you should usually use [filter queries](./filter-queries.md) and similar methods.
+Using Has Checks works best inside the [conversations plugin](../plugins/conversations.md).
```
Comment (roj1512) [inline site/docs/guide/context.md:96]:
```suggestion
Using has checks works best inside the [conversations plugin](../plugins/conversations.md).
```

**Review by roj1512 — APPROVED** [review]

**Comment by CikiMomogi** [comment]
Is it ready for translation?

**Comment by CikiMomogi** [comment]
Apparently you forgot to put the tag on it to notify the translators. I saw that the ES and ZH documentation hasn't been updated yet.

**Comment by roj1512** [comment]
They were notified internally.

**Comment by roj1512** [comment]
Sorry for missing you out.

**Comment by CikiMomogi** [comment]
Santuy

File: site/docs/guide/context.md [hunk 6]
```diff
@@ -83,6 +83,34 @@ bot.on("edited_message", (ctx) => {
...
+::: tip Know When to Use Has Checks
+
+This is the exact same logic that is used by `bot.command("start")`.
+Note that you should usually use [filter queries](./filter-queries.md) and similar methods.
+Using has checks works best inside the [conversations plugin](../plugins/conversations.md).
+
+:::
+
+The has checks correctly narrow down the context type.
+This means that checking if a context has callback query data will tell TypeScript that the context has the field `ctx.callbackQuery.data` present.
```
Comment (CikiMomogi) [inline site/docs/guide/context.md:101]: > This means that checking if a context has callback query data will tell TypeScript that the context has the field `ctx.callbackQuery.data` present.

Would you guys please explain to me what this sentence means? I don't understand the full meaning of the sentence :thinking:

Comment (KnorpelSenf) [reply site/docs/guide/context.md:101]: It means that the has functions are type predicates (https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) which narrow down the type of the context object.

**Pull request merged**